### PR TITLE
Minor CI refactor

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Summary

Refactor of Github Actions CI a bit. See comments for details. 

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting 
(this will be you or your assignee, such as a university or company): 
Rok Češnovar, Univ. of Ljubljana

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ on:
     branches:
       - master
 
-name: R-CMD-check
+name: Unit tests
```
Comment (rok-cesnovar) [inline .github/workflows/R-CMD-check.yaml:9]: Just a nicer name :)

File: .github/workflows/R-CMD-check.yaml [hunk 2]
```diff
@@ -16,7 +16,7 @@ jobs:
     name: ${{ matrix.config.os }} (${{ matrix.config.r }})
 
     strategy:
-      fail-fast: false
+      fail-fast: true
```
Comment (rok-cesnovar) [inline .github/workflows/R-CMD-check.yaml:19]: No need to run all tests if one of them fails. This stops all of them once there is a failure. These are free instances so we don't have to do this, but let's pretend this is saving the environment :)

File: .github/workflows/R-CMD-check.yaml [hunk 3]
```diff
@@ -32,6 +32,12 @@ jobs:
       GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
 
     steps:
+      - uses: n1hility/cancel-previous-runs@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          workflow: R-CMD-check.yaml
+        if: "!startsWith(github.ref, 'refs/tags/') && github.ref != 'refs/heads/master'"
```
Comment (rok-cesnovar) [inline .github/workflows/R-CMD-check.yaml:39]: This cancels previously run jobs for the same PR.

File: .github/workflows/Test-coverage.yaml [hunk 4]
```diff
@@ -48,3 +54,35 @@ jobs:
       - name: Test coverage
         run: covr::codecov(type = "all")
         shell: Rscript {0}
+
+  test-coverage-windows:
```
Comment (rok-cesnovar) [inline .github/workflows/Test-coverage.yaml:58]: Same as above, just for Windows. We special-case a bunch of Windows stuff, so this made sense to me.

**Review by rok-cesnovar — COMMENTED** [review]
Details in comments.

File: .github/workflows/R-CMD-check.yaml [hunk 5]
```diff
@@ -32,8 +32,31 @@ jobs:
...
+
       - uses: actions/checkout@v2
 
+      - name: Set paths
+        if: runner.os == 'Windows'
+        run: |
+          echo "::add-path::C:/rtools40/usr/bin"
+          echo "::add-path::C:/rtools40/mingw64/bin"
+        shell: powershell
```
Comment (rok-cesnovar) [inline .github/workflows/R-CMD-check.yaml:48]: This one and below make sure we use RTools 4.0 toolchain in the tests. That is what we recommend to users and that is what we should test with.

File: R/read_csv.R [hunk 6]
```diff
@@ -214,7 +214,8 @@ read_cmdstan_csv <- function(files,
       altrep = FALSE,
       progress = FALSE,
       skip = metadata$lines_to_skip,
-      col_select = col_select
+      col_select = col_select,
+      num_threads = 1
```
Comment (rok-cesnovar) [inline R/read_csv.R:218]: The macOS tests were still crashing, this fixes that. Will work on removing vroom in favor of something else, so this is temporary.

File: tests/testthat/test-install.R [hunk 7]
```diff
@@ -2,6 +2,7 @@ context("install")
 
 test_that("install_cmdstan() successfully installs cmdstan", {
   skip_if_offline()
+  if (os_is_windows()) skip_on_covr()
```
Comment (rok-cesnovar) [inline tests/testthat/test-install.R:5]: In codecov Windows tests the install tests fail as if g++ can not be found (even though cmdstan is successfully built beforehand). This seems like a weird codecov thing that is not suitable for our install use case, hence I added skips here.

**Review by rok-cesnovar — COMMENTED** [review]
Some more comments.

File: .github/workflows/Test-coverage.yaml (hunk 4 above)
Comment (rok-cesnovar) [reply .github/workflows/Test-coverage.yaml:57]: You can see that this adds 0.74% to test coverage so I think its worth it.

**Review by jgabry — APPROVED** [review]
Looks great. Thanks for the helpful comments!

**Pull request merged**